system_test.go

```diff
@@ -1,287 +1,106 @@
 package main
 
 import (
-	"bytes"
 	"encoding/json"
-	"fmt"
-	"io/ioutil"
 	"os"
-	"os/exec"
-	"sort"
 	"strings"
-	"syscall"
 	"testing"
 	"time"
 
 	"github.com/Shopify/sarama"
-	qt "github.com/frankban/quicktest"
+	"github.com/google/go-cmp/cmp"
+	"github.com/rogpeppe/go-internal/testscript"
 )
 
-type cmd struct {
-	in string
+func TestMain(m *testing.M) {
+	os.Exit(testscript.RunMain(m, map[string]func() int{
+		"kt": func() int {
+			main()
+			// TODO make main return exit code.
+			return 0
+		},
+	}))
 }
 
-func newCmd() *cmd                  { return &cmd{} }
-func (c *cmd) stdIn(in string) *cmd { c.in = in; return c }
-func (c *cmd) run(name string, args ...string) (int, string, string) {
-	cmd := exec.Command(name, args...)
-
-	var stdOut, stdErr bytes.Buffer
-	cmd.Stdout = &stdOut
-	cmd.Stderr = &stdErr
-	cmd.Env = os.Environ()
-	cmd.Env = append(cmd.Env, "KT_BROKERS=localhost:9092")
-
-	if len(c.in) > 0 {
-		cmd.Stdin = strings.NewReader(c.in)
-	}
-
-	_ = cmd.Run()
-	status := cmd.ProcessState.Sys().(syscall.WaitStatus)
-
-	strOut := stdOut.String()
-	strErr := stdErr.String()
-
-	return status.ExitStatus(), strOut, strErr
-
-}
-
-func build(t *testing.T) {
-	c := qt.New(t)
-
-	status, _, _ := newCmd().run("make", "build")
-	c.Assert(status, qt.Equals, 0)
-
-	status, _, _ = newCmd().run("ls", "kt")
-	c.Assert(status, qt.Equals, 0)
-}
+const testBrokerAddr = "localhost:9092"
 
 func TestSystem(t *testing.T) {
-	c := qt.New(t)
-	build(t)
-
-	//
-	// kt admin -createtopic
-	//
-	topicName := fmt.Sprintf("kt-test-%v", randomString(6))
-	topicDetail := &sarama.TopicDetail{
-		NumPartitions:     1,
-		ReplicationFactor: 1,
-	}
-	buf, err := json.Marshal(topicDetail)
-	c.Assert(err, qt.Equals, nil)
-	fnTopicDetail := fmt.Sprintf("topic-detail-%v.json", randomString(6))
-	err = ioutil.WriteFile(fnTopicDetail, buf, 0666)
-	c.Assert(err, qt.Equals, nil)
-	defer os.RemoveAll(fnTopicDetail)
-
-	status, stdOut, stdErr := newCmd().stdIn(string(buf)).run("./kt", "admin", "-createtopic", topicName, "-topicdetail", fnTopicDetail)
-	c.Logf(">> system test kt admin -createtopic %v stdout:\n%s\n", topicName, stdOut)
-	c.Logf(">> system test kt admin -createtopic %v stderr:\n%s\n", topicName, stdErr)
-	c.Assert(status, qt.Equals, 0)
-	c.Assert(stdErr, qt.Equals, "")
-
-	c.Logf(">> ✓\n")
-	//
-	// kt produce
-	//
+	topic := randomString(6)
```

Comment:
testscript is not well know, this could really do with some comment to introduce what its doing.

Reply:
done